### PR TITLE
Fix some typo and update an entry in readme to match the actual code behavior

### PR DESCRIPTION
**Pull request by Realmonia**
### Summary

1. Fix typo
2. The minimum cleanup time is updated in https://github.com/aws/amazon-ecs-agent/pull/3088 but readme is not updated.
### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/ec2/ec2_metadata_client.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (c *ec2MetadataClientImpl) DefaultCredentials() (*RoleCredentials, error) {
 
 	defaultCredentialName := securityCredentialList[0]
 
-	defaultCredentialStr, err := c.client.GetMetadata(SecurityCrednetialsResource + defaultCredentialName)
+	defaultCredentialStr, err := c.client.GetMetadata(SecuritycredentialsResource + defaultCredentialName)
```
Comment (singholt) [inline agent/ec2/ec2_metadata_client.go:120]: `SecurityCredentialResource`?

Comment (singholt) [reply agent/ec2/ec2_metadata_client.go:120]: also, why is it exported do you know?

Comment (Realmonia) [reply agent/ec2/ec2_metadata_client.go:120]: oops, thanks

**Review by amogh09 — APPROVED** [review]

**Review by singholt — APPROVED** [review]

**Review by chienhanlin — APPROVED** [review]

File: agent/ec2/ec2_metadata_client.go (hunk 1 above)
Comment (Realmonia) [reply agent/ec2/ec2_metadata_client.go]: seems exported to be used in test

**Pull request merged**